tests/trainer/test_trainer.py

```diff
@@ -713,5 +716,136 @@ def on_test_end(self, trainer, pl_module):
     assert test_callback.on_test_start_called
     assert test_callback.on_test_end_called
 
+
+def test_optimizer_with_scheduling(tmpdir):
```

Comment:
I would personally move all this addition to new module `tests/trainer/test_optimizers.py`